secure.go

```diff
@@ -101,6 +128,11 @@ func (s *Secure) SetBadHostHandler(handler http.Handler) {
 // Handler implements the http.HandlerFunc for integration with the standard net/http lib.
 func (s *Secure) Handler(h http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if s.opt.nonceEnabled {
+			ctx := context.WithValue(r.Context(), nonceKey, randNonce(rand.NewSource(time.Now().UnixNano()), s.opt.CSPNonceByteAmount))
```

Comment:
Don't use math/rand for this, it's predictable. (Arguably, the odds of pulling off an attack using predictable CSP nonces is low, but why take the risk for security-adjacent code).

Instead, you should probably take an optional "RandSource io.Reader" as one of the options, and if its nil, default to crypto/rand.Reader.

Reply:
To expand on this: use crypto/rand, and *maybe* provide an option to replace it. But you should steer users away from that: it's extremely rare you ever (ever) need to provide your own CSPRNG source.

(In fact, I'd suggest not providing the option, and only adding it if you ever get asked. That's likely to be never.)